src/main/drivers/dshot_bitbang_impl.h

```diff
@@ -0,0 +1,229 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/time.h"
+#include "drivers/motor.h"
+
+#define USE_DMA_REGISTER_CACHE
+
+#define DEBUG_COUNT_INTERRUPT
+#define DEBUG_MONITOR_PACER
+
+#define MAX_SUPPORTED_MOTOR_PORTS 4 // Max direct dshot port groups, limited by number of usable timer (TIM1 and TIM8) x number of channels per timer (4), 3 is enough to cover motor pins on GPIOA, GPIOB and GPIOC.
+
+#define DSHOT_BITBANG_TELEMETRY_OVER_SAMPLE 3
+
+#define DSHOT_BITBANG_DIRECTION_OUTPUT 0
+#define DSHOT_BITBANG_DIRECTION_INPUT  1
+
+#define DSHOT_BITBANG_INVERTED         true
+#define DSHOT_BITBANG_NONINVERTED      false
+
+// XXX MOTOR_xSHOTyyyy_HZ is not usable as generic frequency for timers.
+// XXX Trying to fiddle with constants here.
+
+// Symbol rate [symbol/sec]
+#define MOTOR_DSHOT1200_SYMBOL_RATE    (1200 * 1000)
+#define MOTOR_DSHOT600_SYMBOL_RATE     (600 * 1000)
+#define MOTOR_DSHOT300_SYMBOL_RATE     (300 * 1000)
+#define MOTOR_DSHOT150_SYMBOL_RATE     (150 * 1000)
+#define MOTOR_PROSHOT1000_SYMBOL_RATE  (250 * 1000)
+
+#define MOTOR_DSHOT_SYMBOL_TIME_NS(rate)  (1000000000 / (rate))
+
+#define MOTOR_DSHOT_BIT_PER_SYMBOL         1
+#define MOTOR_PROSHOT1000_BIT_PER_SYMBOL   4
+
+#define MOTOR_DSHOT_STATE_PER_SYMBOL       3  // Initial high, 0/1, low
+#define MOTOR_PROSHOT1000_STATE_PER_SYMBOL 32 // 8 preamble + 16 + 8 postamble
```

Comment:
Proshot is terribly wasteful when using bitbangind, it may be better to drop it)

Reply:
I agree

Reply:
As far as I know, there is no ESC hardware that does ProShot1000 and **with**  RPM feedback (yet). Can we simplify this by disabling bitbanged Dshot if ProShot is selected, and then stripping the ProShot code out of this?

Reply:
blheli32 does proshot1000 with rpm telemetry.

Reply:
There isn't problem with proshot in general. Proshot is good protocol if resources are spent per edge (timer case). But using it with bitbanging is overkill, it is probably OK to force user to avoid this combination.

Reply:
Ok, I should have said 'does well'. @sskaug stated that ProShot1000 has got timing issues on BLHeli_32 hardware.

Reply:
@ledvinap: That's exactly what I suggested - but in this case we should strip the ProShot1000 code from the bitbanged Dshot code as it will only rot anyway if it cannot be used.

Reply:
I removed the 'proshot1000' code and changed 'dshot_bitbang = auto' not to select bitbang with 'proshot1000'. Additionally I changed post config validation to 'set dshot_bitbang = auto' if it's on and proshot1000 is selected.